mlrun/wasbfs/fs.py

```diff
@@ -0,0 +1,104 @@
+import re
+
+from adlfs import AzureBlobFileSystem
+
+
+class WasbFS(AzureBlobFileSystem):
+
+    protocol = "wasb"
+
+    def _convert_wasb_schema_to_az(self, url):
+        # wasbs pattern: wasbs://<CONTAINER>@<ACCOUNT_NAME>.blob.core.windows.net/<PATH_OBJ_IN_CONTAINER>
+        if url.startswith("wasb://") or url.startswith("wasbs://"):
+            m = re.match(
+                r"^(?P<schema>.*)://(?P<cont>.*)@(?P<account>.*?)\..*?/(?P<obj_path>.*?)$",
```

Comment:
I wonder if it wouldn't be more resilient and clearer to use URL parsing (e.g. `urllib.parse.urlparse`)? I can think of a bunch of bad inputs that will cause this to fail with errors or do the wrong thing, such as:
- No match, maybe because the path was left out:
```
AttributeError: 'NoneType' object has no attribute 'groupdict'
```
- Parsing it wrong, maybe because there is a query in the URL? (it would be read in as part of the path)
```
 m.groupdict()
{'schema': 'wasbs', 'cont': 'container', 'account': 'account', 'obj_path': 'path?query=1'}
```

Reply:
Please add unittest for that function.

Reply:
Done